### PR TITLE
feat (engine): option on migrate from old to new wf

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: sdk/cdsclient/deprecated.go [hunk 1]
```diff
@@ -21,6 +21,22 @@ func (c *client) ApplicationPipelinesAttach(projectKey string, appName string, p
 	return nil
 }
 
+func (c *client) ApplicationDoMigrationWorkflow(projectKey string, appName string, force, disablePrefix, withCurrentVersion, withRepositoryWebHook bool) error {
+	uri := fmt.Sprintf("/project/%s/application/%s/workflow/migrate", projectKey, appName)
+	uri += fmt.Sprintf("?force=%t&disablePrefix=%t&withCurrentVersion=%t&withRepositoryWebHook=%t", force, disablePrefix, withCurrentVersion, withRepositoryWebHook)
```
Comment (bnjjj) [inline sdk/cdsclient/deprecated.go]: prefer usage of url and url values https://golang.org/pkg/net/url/#Values

File: ui/src/app/views/application/show/admin/application.admin.html [hunk 2]
```diff
@@ -2,18 +2,44 @@
...
+                            <sm-checkbox label="Disable prefix 'w' in Workflow name generated" [control]="disablePrefix" type="toggle"></sm-checkbox>
+                        </div>
+                        <div class="four field">
+                            <sm-checkbox label="Create Repository Webhook if needed" [control]="withRepositoryWebHook" type="toggle"></sm-checkbox>
+                        </div>
+                        <div class="four field">
+                            <sm-checkbox label="Keep current cds.version" [control]="withCurrentVersion" type="toggle"></sm-checkbox>
+                        </div>                        
+                        <div class="four field">
+                            <div *ngIf="application.workflow_migration == 'STARTED'">
```
Comment (bnjjj) [inline ui/src/app/views/application/show/admin/application.admin.html]: ===

File: ui/src/app/views/application/show/admin/application.admin.html [hunk 3]
```diff
@@ -2,18 +2,44 @@
...
+                                <button class="ui green button" [class.loading]="loading" [disabled]="loading" (click)="generateWorkflow(true)">{{ 'application_migration_regenerate' | translate }}</button>
+                                <button class="ui red button" (click)="openDoneMigrationPopup()">{{ 'application_migration_generate_done' | translate }}</button>
+                            </div>
+                            <div *ngIf="application.workflow_migration != 'STARTED'">
```
Comment (bnjjj) [inline ui/src/app/views/application/show/admin/application.admin.html]: !==

File: engine/api/migrate/migrate_workflow.go [hunk 4]
```diff
@@ -20,10 +20,15 @@ const (
 	STATUS_DONE     = "DONE"
 )
 
-func MigrateToWorkflow(db gorp.SqlExecutor, store cache.Store, cdTree []sdk.CDPipeline, proj *sdk.Project, u *sdk.User, force bool) error {
+func MigrateToWorkflow(db gorp.SqlExecutor, store cache.Store, cdTree []sdk.CDPipeline, proj *sdk.Project, u *sdk.User, force, disablePrefix, withCurrentVersion, withRepositoryWebHook bool) ([]sdk.Workflow, error) {
```
Comment (golangci) [inline engine/api/migrate/migrate_workflow.go]: func name will be used as migrate.MigrateToWorkflow by other packages, and that stutters; consider calling this ToWorkflow

File: engine/api/migrate/migrate_workflow.go [hunk 5]
```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"strings"
 
+	"database/sql"
```
Comment (golangci) [inline engine/api/migrate/migrate_workflow.go]: File is not goimports-ed

**Pull request merged**